test/endpoint_provisioner_test.go

```diff
@@ -0,0 +1,1083 @@
+			Expect(err).To(BeNil())
+
+			// DiscardBehaviour Queue - Now let's provision the endpoint using the API
+			provisioner = provisioner.FromConfigurationProvider(config.EndpointPropertyMap{
+				config.EndpointPropertyDurable:     true,
+				config.EndpointPropertyRespectsTTL: false,
+			})
+			outcome = provisioner.Provision(provisionQueueName, false)
+			Expect(outcome.GetStatus()).To(BeTrue())
+			Expect(outcome.GetError()).ToNot(HaveOccurred())
+
+			// check that the endpoint was provisioned on the broker via semp
+			clientResponse, _, err = testcontext.SEMP().Monitor().MsgVpnApi.
+				GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+			Expect(err).To(BeNil())
+			Expect(clientResponse.Data).ToNot(BeNil())
+			Expect(clientResponse.Data.QueueName).To(Equal(provisionQueueName))
+			Expect(clientResponse.Data.Durable).ToNot(BeNil())              // durability should not be nil
+			Expect(**(clientResponse.Data.Durable)).To(BeTrue())            // durability should be true
+			Expect(clientResponse.Data.RespectTtlEnabled).ToNot(BeNil())    // should not be nil
+			Expect(**(clientResponse.Data.RespectTtlEnabled)).To(BeFalse()) // should be false
+		})
+
+		// TestPlan TestCase Deprovision#4
+		It("deprovision a durable queue which has been deprovisioned before", func() {
+			// Let's first provision the endpoint using the API
+			provisioner = provisioner.FromConfigurationProvider(config.EndpointPropertyMap{
+				config.EndpointPropertyDurable:   false, // this will be overrided to true
+				config.EndpointPropertyExclusive: true,
+			})
+			outcome := provisioner.Provision(provisionQueueName, true)
+			Expect(outcome.GetError()).ToNot(HaveOccurred())
+			Expect(outcome.GetStatus()).To(BeTrue())
+
+			// check that the endpoint is on the broker
+			clientResponse, _, err := testcontext.SEMP().Monitor().MsgVpnApi.
+				GetMsgVpnQueue(testcontext.SEMP().MonitorCtx(), testcontext.Messaging().VPN, url.QueryEscape(provisionQueueName), nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(clientResponse.Data).ToNot(BeNil()) // should have a response
+			Expect(clientResponse.Data.QueueName).To(Equal(provisionQueueName))
+			Expect(**(clientResponse.Data.Durable)).To(BeTrue()) // should have provisioned a durable queue on the broker
+
+			err = provisioner.Deprovision(provisionQueueName, true)
+			Expect(err).To(BeNil()) // should be successful
+
+			err = provisioner.Deprovision(provisionQueueName, true)
+			Expect(err).To(BeNil()) // we should ignore queue missing errors
+
+			err = provisioner.Deprovision(provisionQueueName, false) // call to deprovisioner should be unsuccessful
+			Expect(err).ToNot(BeNil())                               // we should not ignore queue missing errors
+			Expect(string(err.Error())).To(Equal("Unknown Queue"))
+		})
+
+		// TestPlan TestCase Deprovision#8
+		It("should return error from deprovision when queue does not exist on broker", func() {
```

Comment:
How is this test different from the previous? The part you are testing is when there is no queue so why is it different if there used to be one or not?

Reply:
I followed the Test plan document by QA - https://sol-jira.atlassian.net/wiki/spaces/CPD/pages/4031283346/API+s+Support+For+Removing+Durable+Endpoints+Test+Plan . And that is why I added a this comment **"// TestPlan TestCase Deprovision#8"** on the affected test case(s)

Reply:
The test plan says "Try to deprovision a queue(topic endpoint) which does not exists" why is it different if the queue existed at some point before or was de-provisioned by the API already?